workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerHeartbeat/HandlePlayerHeartbeatResponseSystem.cs

```diff
@@ -1,93 +1,93 @@
 
-                foreach (var response in responses)
+                for (int i = 0; i < responses.Length; i++)
                 {
-                    if (response.StatusCode != StatusCode.Success)
+                    var heartbeatData = heartbeats[i];
+
+                    var stillAlive = false;
+                    foreach (var response in responses[i].Responses)
                     {
-                        var failedHeartbeatsComponent = EntityManager.HasComponent<HeartbeatData>(entity)
-                            ? EntityManager.GetComponentData<HeartbeatData>(entity)
-                            : new HeartbeatData();
-                        failedHeartbeatsComponent.NumFailedHeartbeats++;
-                        if (EntityManager.HasComponent<HeartbeatData>(entity))
-                        {
-                            PostUpdateCommands.SetComponent(entity, failedHeartbeatsComponent);
-                        }
-                        else
+                        if (response.StatusCode == StatusCode.Success)
                         {
-                            PostUpdateCommands.AddComponent(entity, failedHeartbeatsComponent);
+                            stillAlive = true;
+                            break;
                         }
+                    }
 
-                        Debug.LogFormat(Messages.FailedHeartbeat, entityId,
-                            failedHeartbeatsComponent.NumFailedHeartbeats,
-                            PlayerLifecycleConfig.MaxNumFailedPlayerHeartbeats);
+                    if (stillAlive)
+                    {
+                        heartbeatData.NumFailedHeartbeats = 0;
+                    }
+                    else
+                    {
+                        heartbeatData.NumFailedHeartbeats += 1;
 
-                        if (failedHeartbeatsComponent.NumFailedHeartbeats >=
-                            PlayerLifecycleConfig.MaxNumFailedPlayerHeartbeats)
+                        if (heartbeatData.NumFailedHeartbeats >= PlayerLifecycleConfig.MaxNumFailedPlayerHeartbeats)
                         {
-                            Debug.LogFormat(Messages.DeletingPlayer, entityId);
+                            var entityDeleteSender = deleteRequesters[i];
                             entityDeleteSender.RequestsToSend.Add(WorldCommands.DeleteEntity.CreateRequest
                             (
-                                entityId
+                                spatialIds[i].EntityId
                             ));
-                            data.WorldCommandSenders[i] = entityDeleteSender;
-                        }
-                    }
-                    else
-                    {
-                        if (EntityManager.HasComponent<HeartbeatData>(entity))
-                        {
-                            PostUpdateCommands.RemoveComponent<HeartbeatData>(entity);
+                            deleteRequesters[i] = entityDeleteSender;
```

Comment:
We don't need to write this back

Reply:
Oops left over from ComponentDataArray, fixed